[Bug] Error raised rdkit from kale\prepdata\chem_transform.py on windows, python 3.10 (with ubuntu no error)

# 🐛 Bug

Test are getting failed because of importing chem from rdkit in kale\prepdata\chem_transform.py on the windows, python 3.10. Proper cause of the error not found yet. Exploring this issue in #377.

## To reproduce
Tests are failed.

** Stack trace/error message **


=================================== ERRORS ====================================
____________ ERROR collecting tests/loaddata/test_tdc_datasets.py _____________
ImportError while importing test module 'D:\a\pykale\pykale\tests\loaddata\test_tdc_datasets.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
C:\Miniconda\envs\test\lib\importlib\__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests\loaddata\test_tdc_datasets.py:4: in <module>
    from kale.loaddata.tdc_datasets import BindingDBDataset
kale\loaddata\tdc_datasets.py:5: in <module>
    from kale.prepdata.chem_transform import integer_label_protein, integer_label_smiles
kale\prepdata\chem_transform.py:9: in <module>
    from rdkit import Chem
C:\Miniconda\envs\test\lib\site-packages\rdkit\Chem\__init__.py:23: in <module>
    from rdkit.Chem.rdmolfiles import *
E   ImportError: DLL load failed while importing rdmolfiles: The specified procedure could not be found.

=========================== short test summary info ===========================
ERROR tests/loaddata/test_tdc_datasets.py
ERROR tests/pipeline/test_deepdta.py
ERROR tests/prepdata/test_chem_transform.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!
======================= 6 warnings, 3 errors in 19.99s ========================

## Expected Behaviour

Tests are passed.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -34,7 +34,8 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          conda install -c conda-forge rdkit
+#          conda install -c conda-forge rdkit
+          pip install rdkit
           pip install pyparsing==2.4.7
           pip install torch==1.11.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
           pip install torchvision==0.12.0+cpu -f https://download.pytorch.org/whl/torch_stable.html

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -34,7 +34,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          conda install -c conda-forge rdkit=2022.03.3
+          conda install -c conda-forge rdkit
           pip install pyparsing==2.4.7
           pip install torch==1.11.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
           pip install torchvision==0.12.0+cpu -f https://download.pytorch.org/whl/torch_stable.html

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -34,7 +34,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          conda install -c conda-forge rdkit=2022.03.2
+          conda install -c conda-forge rdkit=2022.03.3
           pip install pyparsing==2.4.7
           pip install torch==1.11.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
           pip install torchvision==0.12.0+cpu -f https://download.pytorch.org/whl/torch_stable.html

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -34,7 +34,6 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-#          conda install -c conda-forge rdkit
           pip install rdkit
           pip install pyparsing==2.4.7
           pip install torch==1.11.0+cpu -f https://download.pytorch.org/whl/torch_stable.html

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@
 [![tests](https://github.com/pykale/pykale/workflows/test/badge.svg)](https://github.com/pykale/pykale/actions/workflows/test.yml)
 [![codecov](https://codecov.io/gh/pykale/pykale/branch/main/graph/badge.svg?token=jmIYPbA2le)](https://codecov.io/gh/pykale/pykale)
 [![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/optuna/optuna)
-[![Python](https://img.shields.io/badge/python-3.7%20%7C%203.8%20%7C%203.9-blue)](https://www.python.org)
+[![Python](https://img.shields.io/badge/python-3.8%20%7C%203.9%20%7C%203.10-blue)](https://www.python.org)
 [![PyPI version](https://img.shields.io/pypi/v/pykale?color=blue)](https://pypi.org/project/pykale/)
 [![PyPI downloads](https://pepy.tech/badge/pykale)](https://pepy.tech/project/pykale)
 <!-- [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.5557244.svg)](https://doi.org/10.5281/zenodo.5557244) -->
@@ -47,7 +47,7 @@ See our [arXiv preprint](https://arxiv.org/abs/2106.09756) and four short introd
 
 ### Step 0: Installation
 
-PyKale supports Python 3.7, 3.8, or 3.9. Before installing `pykale`, we suggest you to first [install PyTorch](https://pytorch.org/get-started/locally/) matching your hardware, and if graphs will be used, install [PyTorch Geometric](https://github.com/rusty1s/pytorch_geometric) following its [official instructions](https://github.com/rusty1s/pytorch_geometric#installation).
+PyKale supports Python 3.8, 3.9, or 3.10. Before installing `pykale`, we suggest you to first [install PyTorch](https://pytorch.org/get-started/locally/) matching your hardware, and if graphs will be used, install [PyTorch Geometric](https://github.com/rusty1s/pytorch_geometric) following its [official instructions](https://github.com/rusty1s/pytorch_geometric#installation).
 
 Simple installation of `pykale` from [PyPI](https://pypi.org/project/pykale/):
 

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -18,7 +18,7 @@ pandas
 pillow>=6.2.0
 PyTDC
 pytorch-lightning
-rdkit-pypi
+rdkit
 recommonmark
 scikit-learn
 sphinx-markdown-tables

--- a/docs/source/installation.md
+++ b/docs/source/installation.md
@@ -2,7 +2,7 @@
 
 ## Requirements
 
-PyKale requires Python 3.7, 3.8, or 3.9. Before installing pykale, you should
+PyKale requires Python 3.8, 3.9, or 3.10. Before installing pykale, you should
 - manually [install PyTorch](https://pytorch.org/get-started/locally/) matching your hardware first,
 - if you will use APIs related to graphs, you need to manually install [PyTorch Geometric](https://github.com/rusty1s/pytorch_geometric) first following its [official instructions](https://github.com/rusty1s/pytorch_geometric#installation) and matching your PyTorch installation, and
 - If [RDKit](https://www.rdkit.org/) will be used, you need to install it via `conda install -c conda-forge rdkit`.

--- a/docs/source/installation.md
+++ b/docs/source/installation.md
@@ -5,7 +5,7 @@
 PyKale requires Python 3.8, 3.9, or 3.10. Before installing pykale, you should
 - manually [install PyTorch](https://pytorch.org/get-started/locally/) matching your hardware first,
 - if you will use APIs related to graphs, you need to manually install [PyTorch Geometric](https://github.com/rusty1s/pytorch_geometric) first following its [official instructions](https://github.com/rusty1s/pytorch_geometric#installation) and matching your PyTorch installation, and
-- If [RDKit](https://www.rdkit.org/) will be used, you need to install it via `conda install -c conda-forge rdkit`.
+- If [RDKit](https://www.rdkit.org/) will be used, you need to install it via `pip install rdkit`.
 
 ## Pip install
 

--- a/examples/bindingdb_deepdta/README.md
+++ b/examples/bindingdb_deepdta/README.md
@@ -28,7 +28,7 @@ log space.
 You'll need to install the external [RDKit](https://github.com/rdkit/rdkit) package for running the example codes.
 
 ```
-conda install -c conda-forge rdkit
+pip install rdkit
 ```
 
 ### 5. Usage

--- a/examples/bindingdb_deepdta/tutorial.ipynb
+++ b/examples/bindingdb_deepdta/tutorial.ipynb
@@ -1,346 +1,346 @@
 {
-  "nbformat": 4,
-  "nbformat_minor": 5,
-  "metadata": {},
-  "cells": [
-    {
-      "metadata": {},
-      "source": [
-        "# PyKale Tutorial: Drug-Target Interaction Prediction using DeepDTA\n",
-        "\n",
-        "| [Open In Colab](https://colab.research.google.com/github/pykale/pykale/blob/main/examples/bindingdb_deepdta/tutorial.ipynb) (click `Runtime`\u2006\u2192\u2006`Run all (Ctrl+F9)` |\n",
-        "\n",
-        "If using [Google Colab](https://colab.research.google.com), a free GPU can be enabled to save time via setting `Runtime`\u2006\u2192\u2006`Change runtime type` \u2192 `Hardware accelerator: GPU`"
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Introduction\n",
-        "Drug-target interaction prediction is an important research area in the field of drug discovery. It refers to predicting the binding affinity between the given chemical compounds and protein targets. In this example we train a standard DeepDTA model as a baseline in BindingDB, a public, web-accessible dataset of measured binding affinities.\n",
-        "\n",
-        "### DeepDTA\n",
-        "[DeepDTA](https://academic.oup.com/bioinformatics/article/34/17/i821/5093245) is the modeling of protein sequences and compound 1D representations with convolutional neural networks (CNNs). The whole architecture of DeepDTA is shown below.\n",
-        "\n",
-        "![DeepDTA](https://raw.githubusercontent.com/hkmztrk/DeepDTA/master/docs/figures/deepdta.PNG)\n",
-        "\n",
-        "### Datasets\n",
-        "We construct **three datasets** from BindingDB distinguished by different affinity measurement metrics\n",
-        "(**Kd, IC50 and Ki**). They are acquired from [Therapeutics Data Commons](https://tdcommons.ai/) (TDC), which is a collection of machine learning tasks spreading across different domains of therapeutics. The data statistics is shown below:\n",
-        "\n",
-        "|  Metrics   | Drugs | Targets | Pairs |\n",
-        "|  :----:  | :----:  |   :----:  | :----:  |\n",
-        "| Kd  | 10,655 | 1,413 | 52,284 |\n",
-        "| IC50  | 549,205 | 5,078 | 991,486 |\n",
-        "| Ki | 174,662 | 3,070 | 375,032 |\n",
-        "\n",
-        "This figure is the binding affinity distribution for the three datasets respectively, where the metric values (x-axis) have been transformed into log space.\n",
-        "![Binding affinity distribution](figures/bindingdb.jpg)\n",
-        "This tutorial uses the (smallest) **Kd** dataset.\n",
-        "\n",
-        "## Setup\n",
-        "\n",
-        "The first few blocks of code are necessary to set up the notebook execution environment and import the required modules, including PyKale.\n",
-        "\n",
-        "This checks if the notebook is running on Google Colab and installs required packages."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "if 'google.colab' in str(get_ipython()):\n",
-        "    print('Running on CoLab')\n",
-        "    !pip uninstall --yes imgaug && pip uninstall --yes albumentations && pip install git+https://github.com/aleju/imgaug.git\n",
-        "    !pip install rdkit-pypi\n",
-        "    !git clone https://github.com/pykale/pykale.git\n",
-        "    %cd pykale\n",
-        "    !pip install .[graph,example] \n",
-        "    %cd examples/bindingdb_deepdta\n",
-        "else:\n",
-        "    print('Not running on CoLab')"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "import os\n",
-        "import torch\n",
-        "os.environ['TORCH'] = torch.__version__\n",
-        "print(torch.__version__)\n",
-        "\n",
-        "!pip install -q torch-scatter -f https://data.pyg.org/whl/torch-${TORCH}.html\n",
-        "!pip install -q torch-sparse -f https://data.pyg.org/whl/torch-${TORCH}.html\n",
-        "!pip install -q git+https://github.com/pyg-team/pytorch_geometric.git"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "This imports required modules."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "import pytorch_lightning as pl\n",
-        "import torch\n",
-        "from config import get_cfg_defaults\n",
-        "from model import get_model\n",
-        "from pytorch_lightning.callbacks import ModelCheckpoint\n",
-        "from pytorch_lightning.loggers import TensorBoardLogger\n",
-        "from torch.utils.data import DataLoader, Subset\n",
-        "\n",
-        "from kale.loaddata.tdc_datasets import BindingDBDataset\n",
-        "from kale.utils.seed import set_seed"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Configuration\n",
-        "\n",
-        "The customized configuration used in this tutorial is stored in `./configs/tutorial.yaml`, this file overwrites defaults in `config.py` where a value is specified.\n",
-        "\n",
-        "For saving time to run a whole pipeline in this tutorial, we sample small train/valid/test (8,000/1,000/1,000) subsets from the original BindingDB dataset."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "cfg_path = \"./configs/tutorial.yaml\"\n",
-        "train_subset_size, valid_subset_size, test_subset_size = 8000, 1000, 1000\n",
-        "\n",
-        "cfg = get_cfg_defaults()\n",
-        "cfg.merge_from_file(cfg_path)\n",
-        "cfg.freeze()\n",
-        "print(cfg)\n",
-        "\n",
-        "set_seed(cfg.SOLVER.SEED)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Check if a GPU is available\n",
-        "\n",
-        "If a CUDA GPU is available, this should be used to accelerate the training process. The code below checks and reports on this.\n"
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
-        "print(\"Using: \" + device)\n",
-        "gpus = 1 if device == \"cuda\" else 0"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Select Datasets\n",
-        "\n",
-        "Source and target datasets are specified using the `BindingDBDataset()` function and loaded using the `DataLoader()` function."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "train_dataset = BindingDBDataset(name=cfg.DATASET.NAME, split=\"train\", path=cfg.DATASET.PATH)\n",
-        "valid_dataset = BindingDBDataset(name=cfg.DATASET.NAME, split=\"valid\", path=cfg.DATASET.PATH)\n",
-        "test_dataset = BindingDBDataset(name=cfg.DATASET.NAME, split=\"test\", path=cfg.DATASET.PATH)\n",
-        "train_size, valid_size, test_size = len(train_dataset), len(valid_dataset), len(test_dataset)\n",
-        "train_sample_indices, valid_sample_indices, test_sample_indices = torch.randperm(train_size)[:train_subset_size].tolist(), torch.randperm(valid_size)[:valid_subset_size].tolist(), torch.randperm(test_size)[:test_subset_size].tolist()\n",
-        "train_dataset, valid_dataset, test_dataset = Subset(train_dataset, train_sample_indices), Subset(valid_dataset, valid_sample_indices), Subset(test_dataset, test_sample_indices)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "cfg.DATASET.PATH"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "train_loader = DataLoader(dataset=train_dataset, shuffle=True, batch_size=cfg.SOLVER.TRAIN_BATCH_SIZE)\n",
-        "valid_loader = DataLoader(dataset=valid_dataset, shuffle=True, batch_size=cfg.SOLVER.TEST_BATCH_SIZE)\n",
-        "test_loader = DataLoader(dataset=test_dataset, shuffle=True, batch_size=cfg.SOLVER.TEST_BATCH_SIZE)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Setup model\n",
-        "\n",
-        "Here, we use the previously defined configuration and dataset to set up the model we will subsequently train."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "model = get_model(cfg)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Setup Logger\n",
-        "\n",
-        "A logger is used to store output generated during and after model training. This information can be used to assess the effectiveness of the training and to identify problems."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "tb_logger = TensorBoardLogger(\"tb_logs\", name=cfg.DATASET.NAME)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Setup Trainer\n",
-        "\n",
-        "A trainer object is used to determine and store model parameters. Here, one is configured with information on how a model should be trained, and what hardware will be used."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "checkpoint_callback = ModelCheckpoint(filename=\"{epoch}-{step}-{valid_loss:.4f}\", monitor=\"valid_loss\", mode=\"min\")\n",
-        "trainer = pl.Trainer(min_epochs=cfg.SOLVER.MIN_EPOCHS, \n",
-        "                     max_epochs=cfg.SOLVER.MAX_EPOCHS, \n",
-        "                     gpus=gpus, logger=tb_logger, \n",
-        "                     callbacks=[checkpoint_callback])"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Train Model\n",
-        "\n",
-        "Optimize model parameters using the trainer."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "%time trainer.fit(model, train_dataloaders=train_loader, val_dataloaders=valid_loader)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Test Optimized Model\n",
-        "\n",
-        "Check performance of model optimized with training data against test data which was not used in training."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "trainer.test(dataloaders=test_loader)"
-      ],
-      "cell_type": "code",
-      "outputs": [],
-      "execution_count": null
-    },
-    {
-      "metadata": {},
-      "source": [
-        "You should get a test loss of $7.3\\cdots$ in root mean square error (RMSE). The target value ($y$) has a range of [-13, 20] (in log space). Thus, with only three epochs, we have learned to predict the target value with an RMSE of 7.3 over a range of [-13, 20].\n",
-        "\n",
-        "We set the maximum epochs to 3 and extract a subset (8000/1000/1000) to save time in running this tutorial. You may change these settings. Setting the max epochs to 100 and using the full dataset will get a much better result (<1)."
-      ],
-      "cell_type": "markdown"
-    },
-    {
-      "metadata": {},
-      "source": [
-        "## Architecture\n",
-        "Below is the architecture of DeepDTA with default hyperparameters settings.\n",
-        "\n",
-        "<pre>\n",
-        "==========================================================================================\n",
-        "Layer (type:depth-idx)                   Output Shape              Param #\n",
-        "==========================================================================================\n",
-        "\u251c\u2500CNNEncoder: 1-1                        [256, 96]                 --\n",
-        "|    \u2514\u2500Embedding: 2-1                    [256, 85, 128]            8,320\n",
-        "|    \u2514\u2500Conv1d: 2-2                       [256, 32, 121]            21,792\n",
-        "|    \u2514\u2500Conv1d: 2-3                       [256, 64, 114]            16,448\n",
-        "|    \u2514\u2500Conv1d: 2-4                       [256, 96, 107]            49,248\n",
-        "|    \u2514\u2500AdaptiveMaxPool1d: 2-5            [256, 96, 1]              --\n",
-        "\u251c\u2500CNNEncoder: 1-2                        [256, 96]                 --\n",
-        "|    \u2514\u2500Embedding: 2-6                    [256, 1200, 128]          3,328\n",
-        "|    \u2514\u2500Conv1d: 2-7                       [256, 32, 121]            307,232\n",
-        "|    \u2514\u2500Conv1d: 2-8                       [256, 64, 114]            16,448\n",
-        "|    \u2514\u2500Conv1d: 2-9                       [256, 96, 107]            49,248\n",
-        "|    \u2514\u2500AdaptiveMaxPool1d: 2-10           [256, 96, 1]              --\n",
-        "\u251c\u2500MLPDecoder: 1-3                        [256, 1]                  --\n",
-        "|    \u2514\u2500Linear: 2-11                      [256, 1024]               197,632\n",
-        "|    \u2514\u2500Dropout: 2-12                     [256, 1024]               --\n",
-        "|    \u2514\u2500Linear: 2-13                      [256, 1024]               1,049,600\n",
-        "|    \u2514\u2500Dropout: 2-14                     [256, 1024]               --\n",
-        "|    \u2514\u2500Linear: 2-15                      [256, 512]                524,800\n",
-        "|    \u2514\u2500Linear: 2-16                      [256, 1]                  513\n",
-        "==========================================================================================\n",
-        "Total params: 2,244,609\n",
-        "Trainable params: 2,244,609\n",
-        "Non-trainable params: 0\n",
-        "Total mult-adds (M): 58.08\n",
-        "==========================================================================================\n",
-        "Input size (MB): 1.32\n",
-        "Forward/backward pass size (MB): 429.92\n",
-        "Params size (MB): 8.98\n",
-        "Estimated Total Size (MB): 440.21"
-      ],
-      "cell_type": "markdown"
-    }
-  ]
+ "nbformat": 4,
+ "nbformat_minor": 5,
+ "metadata": {},
+ "cells": [
+  {
+   "metadata": {},
+   "source": [
+    "# PyKale Tutorial: Drug-Target Interaction Prediction using DeepDTA\n",
+    "\n",
+    "| [Open In Colab](https://colab.research.google.com/github/pykale/pykale/blob/main/examples/bindingdb_deepdta/tutorial.ipynb) (click `Runtime` → `Run all (Ctrl+F9)` |\n",
+    "\n",
+    "If using [Google Colab](https://colab.research.google.com), a free GPU can be enabled to save time via setting `Runtime` → `Change runtime type` → `Hardware accelerator: GPU`"
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "Drug-target interaction prediction is an important research area in the field of drug discovery. It refers to predicting the binding affinity between the given chemical compounds and protein targets. In this example we train a standard DeepDTA model as a baseline in BindingDB, a public, web-accessible dataset of measured binding affinities.\n",
+    "\n",
+    "### DeepDTA\n",
+    "[DeepDTA](https://academic.oup.com/bioinformatics/article/34/17/i821/5093245) is the modeling of protein sequences and compound 1D representations with convolutional neural networks (CNNs). The whole architecture of DeepDTA is shown below.\n",
+    "\n",
+    "![DeepDTA](https://raw.githubusercontent.com/hkmztrk/DeepDTA/master/docs/figures/deepdta.PNG)\n",
+    "\n",
+    "### Datasets\n",
+    "We construct **three datasets** from BindingDB distinguished by different affinity measurement metrics\n",
+    "(**Kd, IC50 and Ki**). They are acquired from [Therapeutics Data Commons](https://tdcommons.ai/) (TDC), which is a collection of machine learning tasks spreading across different domains of therapeutics. The data statistics is shown below:\n",
+    "\n",
+    "|  Metrics   | Drugs | Targets | Pairs |\n",
+    "|  :----:  | :----:  |   :----:  | :----:  |\n",
+    "| Kd  | 10,655 | 1,413 | 52,284 |\n",
+    "| IC50  | 549,205 | 5,078 | 991,486 |\n",
+    "| Ki | 174,662 | 3,070 | 375,032 |\n",
+    "\n",
+    "This figure is the binding affinity distribution for the three datasets respectively, where the metric values (x-axis) have been transformed into log space.\n",
+    "![Binding affinity distribution](figures/bindingdb.jpg)\n",
+    "This tutorial uses the (smallest) **Kd** dataset.\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "The first few blocks of code are necessary to set up the notebook execution environment and import the required modules, including PyKale.\n",
+    "\n",
+    "This checks if the notebook is running on Google Colab and installs required packages."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "if 'google.colab' in str(get_ipython()):\n",
+    "    print('Running on CoLab')\n",
+    "    !pip uninstall --yes imgaug && pip uninstall --yes albumentations && pip install git+https://github.com/aleju/imgaug.git\n",
+    "    !pip install rdkit\n",
+    "    !git clone https://github.com/pykale/pykale.git\n",
+    "    %cd pykale\n",
+    "    !pip install .[graph,example] \n",
+    "    %cd examples/bindingdb_deepdta\n",
+    "else:\n",
+    "    print('Not running on CoLab')"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "import os\n",
+    "import torch\n",
+    "os.environ['TORCH'] = torch.__version__\n",
+    "print(torch.__version__)\n",
+    "\n",
+    "!pip install -q torch-scatter -f https://data.pyg.org/whl/torch-${TORCH}.html\n",
+    "!pip install -q torch-sparse -f https://data.pyg.org/whl/torch-${TORCH}.html\n",
+    "!pip install -q git+https://github.com/pyg-team/pytorch_geometric.git"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "This imports required modules."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "import pytorch_lightning as pl\n",
+    "import torch\n",
+    "from config import get_cfg_defaults\n",
+    "from model import get_model\n",
+    "from pytorch_lightning.callbacks import ModelCheckpoint\n",
+    "from pytorch_lightning.loggers import TensorBoardLogger\n",
+    "from torch.utils.data import DataLoader, Subset\n",
+    "\n",
+    "from kale.loaddata.tdc_datasets import BindingDBDataset\n",
+    "from kale.utils.seed import set_seed"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Configuration\n",
+    "\n",
+    "The customized configuration used in this tutorial is stored in `./configs/tutorial.yaml`, this file overwrites defaults in `config.py` where a value is specified.\n",
+    "\n",
+    "For saving time to run a whole pipeline in this tutorial, we sample small train/valid/test (8,000/1,000/1,000) subsets from the original BindingDB dataset."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "cfg_path = \"./configs/tutorial.yaml\"\n",
+    "train_subset_size, valid_subset_size, test_subset_size = 8000, 1000, 1000\n",
+    "\n",
+    "cfg = get_cfg_defaults()\n",
+    "cfg.merge_from_file(cfg_path)\n",
+    "cfg.freeze()\n",
+    "print(cfg)\n",
+    "\n",
+    "set_seed(cfg.SOLVER.SEED)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Check if a GPU is available\n",
+    "\n",
+    "If a CUDA GPU is available, this should be used to accelerate the training process. The code below checks and reports on this.\n"
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
+    "print(\"Using: \" + device)\n",
+    "gpus = 1 if device == \"cuda\" else 0"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Select Datasets\n",
+    "\n",
+    "Source and target datasets are specified using the `BindingDBDataset()` function and loaded using the `DataLoader()` function."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "train_dataset = BindingDBDataset(name=cfg.DATASET.NAME, split=\"train\", path=cfg.DATASET.PATH)\n",
+    "valid_dataset = BindingDBDataset(name=cfg.DATASET.NAME, split=\"valid\", path=cfg.DATASET.PATH)\n",
+    "test_dataset = BindingDBDataset(name=cfg.DATASET.NAME, split=\"test\", path=cfg.DATASET.PATH)\n",
+    "train_size, valid_size, test_size = len(train_dataset), len(valid_dataset), len(test_dataset)\n",
+    "train_sample_indices, valid_sample_indices, test_sample_indices = torch.randperm(train_size)[:train_subset_size].tolist(), torch.randperm(valid_size)[:valid_subset_size].tolist(), torch.randperm(test_size)[:test_subset_size].tolist()\n",
+    "train_dataset, valid_dataset, test_dataset = Subset(train_dataset, train_sample_indices), Subset(valid_dataset, valid_sample_indices), Subset(test_dataset, test_sample_indices)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "cfg.DATASET.PATH"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "train_loader = DataLoader(dataset=train_dataset, shuffle=True, batch_size=cfg.SOLVER.TRAIN_BATCH_SIZE)\n",
+    "valid_loader = DataLoader(dataset=valid_dataset, shuffle=True, batch_size=cfg.SOLVER.TEST_BATCH_SIZE)\n",
+    "test_loader = DataLoader(dataset=test_dataset, shuffle=True, batch_size=cfg.SOLVER.TEST_BATCH_SIZE)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Setup model\n",
+    "\n",
+    "Here, we use the previously defined configuration and dataset to set up the model we will subsequently train."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "model = get_model(cfg)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Setup Logger\n",
+    "\n",
+    "A logger is used to store output generated during and after model training. This information can be used to assess the effectiveness of the training and to identify problems."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "tb_logger = TensorBoardLogger(\"tb_logs\", name=cfg.DATASET.NAME)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Setup Trainer\n",
+    "\n",
+    "A trainer object is used to determine and store model parameters. Here, one is configured with information on how a model should be trained, and what hardware will be used."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "checkpoint_callback = ModelCheckpoint(filename=\"{epoch}-{step}-{valid_loss:.4f}\", monitor=\"valid_loss\", mode=\"min\")\n",
+    "trainer = pl.Trainer(min_epochs=cfg.SOLVER.MIN_EPOCHS, \n",
+    "                     max_epochs=cfg.SOLVER.MAX_EPOCHS, \n",
+    "                     gpus=gpus, logger=tb_logger, \n",
+    "                     callbacks=[checkpoint_callback])"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Train Model\n",
+    "\n",
+    "Optimize model parameters using the trainer."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "%time trainer.fit(model, train_dataloaders=train_loader, val_dataloaders=valid_loader)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Test Optimized Model\n",
+    "\n",
+    "Check performance of model optimized with training data against test data which was not used in training."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "trainer.test(dataloaders=test_loader)"
+   ],
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "source": [
+    "You should get a test loss of $7.3\\cdots$ in root mean square error (RMSE). The target value ($y$) has a range of [-13, 20] (in log space). Thus, with only three epochs, we have learned to predict the target value with an RMSE of 7.3 over a range of [-13, 20].\n",
+    "\n",
+    "We set the maximum epochs to 3 and extract a subset (8000/1000/1000) to save time in running this tutorial. You may change these settings. Setting the max epochs to 100 and using the full dataset will get a much better result (<1)."
+   ],
+   "cell_type": "markdown"
+  },
+  {
+   "metadata": {},
+   "source": [
+    "## Architecture\n",
+    "Below is the architecture of DeepDTA with default hyperparameters settings.\n",
+    "\n",
+    "<pre>\n",
+    "==========================================================================================\n",
+    "Layer (type:depth-idx)                   Output Shape              Param #\n",
+    "==========================================================================================\n",
+    "├─CNNEncoder: 1-1                        [256, 96]                 --\n",
+    "|    └─Embedding: 2-1                    [256, 85, 128]            8,320\n",
+    "|    └─Conv1d: 2-2                       [256, 32, 121]            21,792\n",
+    "|    └─Conv1d: 2-3                       [256, 64, 114]            16,448\n",
+    "|    └─Conv1d: 2-4                       [256, 96, 107]            49,248\n",
+    "|    └─AdaptiveMaxPool1d: 2-5            [256, 96, 1]              --\n",
+    "├─CNNEncoder: 1-2                        [256, 96]                 --\n",
+    "|    └─Embedding: 2-6                    [256, 1200, 128]          3,328\n",
+    "|    └─Conv1d: 2-7                       [256, 32, 121]            307,232\n",
+    "|    └─Conv1d: 2-8                       [256, 64, 114]            16,448\n",
+    "|    └─Conv1d: 2-9                       [256, 96, 107]            49,248\n",
+    "|    └─AdaptiveMaxPool1d: 2-10           [256, 96, 1]              --\n",
+    "├─MLPDecoder: 1-3                        [256, 1]                  --\n",
+    "|    └─Linear: 2-11                      [256, 1024]               197,632\n",
+    "|    └─Dropout: 2-12                     [256, 1024]               --\n",
+    "|    └─Linear: 2-13                      [256, 1024]               1,049,600\n",
+    "|    └─Dropout: 2-14                     [256, 1024]               --\n",
+    "|    └─Linear: 2-15                      [256, 512]                524,800\n",
+    "|    └─Linear: 2-16                      [256, 1]                  513\n",
+    "==========================================================================================\n",
+    "Total params: 2,244,609\n",
+    "Trainable params: 2,244,609\n",
+    "Non-trainable params: 0\n",
+    "Total mult-adds (M): 58.08\n",
+    "==========================================================================================\n",
+    "Input size (MB): 1.32\n",
+    "Forward/backward pass size (MB): 429.92\n",
+    "Params size (MB): 8.98\n",
+    "Estimated Total Size (MB): 440.21"
+   ],
+   "cell_type": "markdown"
+  }
+ ]
 }

--- a/examples/video_loading/README.md
+++ b/examples/video_loading/README.md
@@ -67,7 +67,7 @@ for image in frames:
 # Without these three, VideoFrameDataset will not work.
 torchvision >= 0.10.0
 torch >= 1.9.0
-python >= 3.7,<3.10
+python >= 3.8,<3.11
 ```
 
 ### 2. Custom Dataset

--- a/kale/embed/factorization.py
+++ b/kale/embed/factorization.py
@@ -375,7 +375,7 @@ class MIDA(BaseEstimator, TransformerMixin):
 
         self.eig_values_ = eig_values[idx_sorted]
         self.U = eig_vectors[:, idx_sorted]
-        self.U = np.asarray(self.U, dtype=np.float)
+        self.U = np.asarray(self.U, dtype=np.float64)
 
         return self
 

--- a/kale/loaddata/sampler.py
+++ b/kale/loaddata/sampler.py
@@ -182,7 +182,7 @@ class ReweightedBatchSampler(torch.utils.data.sampler.BatchSampler):
             krem = k - n_classes
             wrem = 1 - sum_w
             logging.warning(f"will assume uniform distribution for labels > {len(class_weights)}")
-            self._class_weights = np.ones(n_classes, dtype=np.float)
+            self._class_weights = np.ones(n_classes, dtype=np.float64)
             self._class_weights[:k] = class_weights
             self._class_weights[k:] = wrem / krem
         else:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ requires = [
 [tool.black]
 # https://github.com/psf/black
 line-length = 120
-target-version = ["py37"]
+target-version = ["py38"]
 
 [tool.isort]
 known_first_party = [

